### ISSUE TITLE
[feature request] F.interpolate_as

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#52845]
To make easier computation of shape parameter for `F.interpolate`

I stumbled on this need when coding an auto-encoder (given that padding / strides are hard to compute properly)
